integration-tests/test-oncotree-dependent-tools.sh

```diff
@@ -0,0 +1,115 @@
+ONCOTREE_DEPLOYMENT_SUCCESS=0
+CURRENT_WAIT_TIME=0
+MAXIMUM_WAIT_TIME=600
+ONCOTREE_URL="http://dashi-dev.cbio.mskcc.org:$ONCOTREE_PORT"
+while [ $ONCOTREE_DEPLOYMENT_SUCCESS -eq 0 ] ; do
+    CURRENT_WAIT_TIME=$(($CURRENT_WAIT_TIME + 10))
+    JOB_RUNNING=`jobs | grep "oncotree.jar" | grep "Running"`
+    if [ -z $JOB_RUNNING ] ; then
+        curl --fail -X GET --header 'Accept: */*' $ONCOTREE_URL       
+        if [ $? -eq 0 ] ; then
+            ONCOTREE_DEPLOYMENT_SUCCESS=1
+            break
+        fi
+    else
+        echo "ONCOTREE unable to start up... canceling tests"
+        break
+    fi
+    
+    if [ $CURRENT_WAIT_TIME -gt $MAXIMUM_WAIT_TIME ] ; then
+        echo "ONCOTREE is inaccessible (after a 10 min wait time)... canceling tests"
+    fi 
+    sleep 10
+done
+
+if [ $ONCOTREE_DEPLOYMENT_SUCCESS -gt 0 ] ; then    
+    cp $ONCOTREE_DIRECTORY/integration-test/default_oncotree_file.txt $TESTING_DIRECTORY_TEMP/test_oncotree_default.txt
+    cp $ONCOTREE_DIRECTORY/integration-test/default_oncotree_file.txt $TESTING_DIRECTORY_TEMP/test_oncotree_version.txt
+    
+    # add headers using test ONCOTREE - failure means new ONCOTREE schema not compatible (i.e invalid endpoint, invalid returned json)
+    python $IMPORT_SCRIPTS_DIRECTORY/oncotree_code_converter.py -c $TESTING_DIRECTORY_TEMP/test_oncotree_default.txt -o $ONCOTREE_URL
+    if [ $? -gt 0 ] ; then
+        echo "call to test ONCOTREE failed -- new ONCOTREE code incompatible with CMO pipelines (oncotree_code_converter.py)"
+    else
+        ONCOTREE_CODE_CONVERTER_TEST_SUCCESS=1
+    fi
+    
+    python $IMPORT_SCRIPTS_DIRECTORY/oncotree_code_converter.py -c $TESTING_DIRECTORY_TEMP/test_oncotree_version.txt -o $ONCOTREE_URL -v invalid_oncotree_versione
+    if [ $? -eq 0 ] ; then
+        echo "call to test ONCOTREE with invalid version succeeded -- new ONCOTREE code incompatible with CMO pipelines (oncotree_code_converter.py)"
+    else
+        FAKE_ONCOTREE_VERSION_TEST_SUCCESS=1
+    fi
+    
+    python $IMPORT_SCRIPTS_DIRECTORY/oncotree_code_converter.py -c $TESTING_DIRECTORY_TEMP/test_oncotree_version.txt -o $ONCOTREE_URL -v oncotree_candidate_releasee
+    python verify_oncotree_code_converter_output.py -f $TESTING_DIRECTORY_TEMP/test_oncotree_version.txt  
+    if [ $? -gt 0 ] ; then
+        echo "call to test ONCOTREE resulted in unexpected output -- new ONCOTREE code incompatible with CMO pipelines (oncotree_code_converter.py)"
+    else
+        ONCOTREE_CODE_CONVERTER_OUTPUT_TEST_SUCCESS=1
+    fi
+
+    # kill test ONCOTREE process
+    ONCOTREE_PROCESS_NUMBER=`find_oncotree_process_number $ONCOTREE_PORT`
+    kill $ONCOTREE_PROCESS_NUMBER
+fi
+ 
+rm -rf $TESTING_DIRECTORY_TEMP
+
+# all four tests must pass for integration test to succeed 
+if [[ $ONCOTREE_CODE_CONVERTER_TEST_SCCESS -eq 0 || $FAKE_ONCOTREE_VERSION_TEST_SUCCESS -eq 0 || $ONCOTREE_CODE_CONVERTER_OUTPUT_TEST_SUCCESS -eq 0 ]] ; then
```

Comment:
shouldn't the logic be "A and B and C" rather than "A or B or C" ?
Also, the comment says four tests, but there are three conditions being checked here.

Reply:
the SUCCESS flag is set to 1 if the individual test succeeds (0 indicates failure). 
Using "or" here means we will exit with a non-zero exit status (failed build in Jenkins) if any of the tests fail.